File: DVLD_Project/TestMainInformations.vb
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports BussnessLayerForDVLD

Namespace DVLD_Project
    Public Partial Class TestMainInformations
        Inherits UserControl
        Public Sub New()
            InitializeComponent()
        End Sub


        Public Sub LoadControal(Mode As String)
            If Equals(Mode, "Read") Then
                dateTimePicker1.Visible = False
                lblTestDate.Visible = True
            Else
                dateTimePicker1.Visible = True
                lblTestDate.Visible = False
            End If
        End Sub

        Public Sub LoadControal(Mode As String, ExamMode As Integer, DL As Integer, TestID As Integer, ClassID As Integer)
            Dim dt = clsBussnesLayer.GetInfoTest(DL, TestID, ClassID)
            If dt IsNot Nothing Then
                For Each dr As DataRow In dt.Rows

                    lblAppID.Text = dr("LocalDrivingLicenseApplicationID").ToString()
                    lblName.Text = dr("FullName").ToString()
                    lblType.Text = dr("ClassName").ToString()
                    lblFees.Text = "10"
                    lblTrye.Text = dr("Trayes").ToString()
                Next
            End If
            If Equals(Mode, "Read") Then
                dateTimePicker1.Visible = False
                lblTestDate.Visible = True
            Else
                dateTimePicker1.Visible = True
                lblTestDate.Visible = False
            End If

        End Sub

        Public Function GetTotalTryes() As Integer
            Return Convert.ToInt32(lblTrye.Text)
        End Function

        Private Sub TestMainInformations_Load(sender As Object, e As EventArgs)

        End Sub
        Public Function GetNewDateTime() As Date
            Return dateTimePicker1.Value
        End Function
    End Class
End Namespace

File: DVLD_Project/TestMainInformations.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using BussnessLayerForDVLD;

namespace DVLD_Project
{
    public partial class TestMainInformations : UserControl
    {
        public TestMainInformations()
        {
            InitializeComponent();
        }


        public void LoadControal(string Mode)
        {
            if (Mode == "Read")
            {
                dateTimePicker1.Visible = false;
                lblTestDate.Visible = true;
            }
            else
            {
                dateTimePicker1.Visible = true;
                lblTestDate.Visible = false;
            }
        }

        public void LoadControal(string Mode,int ExamMode,int DL,int TestID,int ClassID)
        {
            DataTable dt = clsBussnesLayer.GetInfoTest(DL,TestID,ClassID);
            if (dt != null)
            {
                foreach (DataRow dr in dt.Rows)
                {

                    lblAppID.Text = dr["LocalDrivingLicenseApplicationID"].ToString();
                    lblName.Text = dr["FullName"].ToString();
                    lblType.Text= dr["ClassName"].ToString();
                    lblFees.Text = "10";
                    lblTrye.Text = dr["Trayes"].ToString();
                }
            }
            if (Mode == "Read")
            {
                dateTimePicker1.Visible = false;
                lblTestDate.Visible = true;
            }
            else
            {
                dateTimePicker1.Visible = true;
                lblTestDate.Visible = false;
            }

        }

        public int GetTotalTryes()
        {
            int Num = 0;
            int.TryParse(lblTrye.Text, out Num);
            return  Num;
        }

        private void TestMainInformations_Load(object sender, EventArgs e)
        {

        }
        public DateTime GetNewDateTime()
        {
            return dateTimePicker1.Value;
        }
    }
}
